### PR TITLE
Changes for AIX platform

### PR DESCRIPTION
**Pull request by kavanabhat**
Changes for AIX platform

### CODE REVIEW

**Comment by prasanthpul** [comment]
Is there a CI for this configuration so we can tell if something ever breaks?

**Comment by kavanabhat** [comment]
Existing CI tests will suffice for AIX too.

**Comment by prasanthpul** [comment]
Hi @kavanabhat , the concern is not about the tests but rather having a pipeline instance that runs on that platform and exercises the code path. Could you setup a CI pipeline like was done with PPC (http://powerci.osuosl.org/job/onnx-ppc64le-nightly-build/)?

**Comment by kavanabhat** [comment]
We are looking into this and will need more time to have CI pipeline setup on AIX. Let us know if there are any guidelines on setting this up. Till then we plan to internally test new releases and report if any issues to the community. If the changes look fine, we can have the code merged.

**Comment by kavanabhat** [comment]
Please let me know if we can have these changes merged. Thanks.

**Comment by kavanabhat** [comment]
@houseroad and @linkerzhang , let me know if the changes look ok. Thanks.

File: CMakeLists.txt [hunk 1]
```diff
@@ -331,7 +335,14 @@ else()
 endif()
 add_onnx_global_defines(onnx_proto)
 
-add_library(onnx ${ONNX_SRCS})
+if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
+  #whole-archive linker option not available on AIX.
```
Comment (houseroad) [inline CMakeLists.txt]: lint the code? space after #

**Comment by kavanabhat** [comment]
@houseroad, the suggested changes have been incorporated. Please have these changes merged.

**Comment by houseroad** [comment]
@kavanabhat sure, any eta for the AIX CI btw? :-)

**Comment by kavanabhat** [comment]
As mentioned earlier, will need more time to have CI pipeline setup on AIX. Let us know if there are any guidelines on setting this up. Till then we plan to internally test new releases and report if any issues to the community.

**Comment by prasanthpul** [comment]
Folks - should we merge this for 1.6?

**Comment by prasanthpul** [comment]
@kavanabhat is the CI now setup?

**Comment by houseroad** [comment]
Yeah, code looks good. Merged :-)

**Pull request merged**